js/index.js

```diff
@@ -1,23 +1,35 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
+  let price = product.querySelector('.price span').innerHTML;
+  let quantity = product.querySelector('.quantity input').value;
+  let subTotal = price * quantity;
 
-  //... your code goes here
+  let subTotalElm = product.querySelector('.subtotal span');
+  subTotalElm.innerText = subTotal;
+
+  return subTotal;
 }
 
 function calculateAll() {
   // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
+  // const singleProduct = document.querySelector('.product');
+  // updateSubtotal(singleProduct);
   // end of test
 
-  // ITERATION 2
-  //... your code goes here
+  // ITERATION 2 & 3
 
-  // ITERATION 3
-  //... your code goes here
+  let total = 0;
+
+  const allProductList = document.querySelectorAll('.product');
+
+  for (let i = 0; i < allProductList.length; i++) {
+    total += updateSubtotal(allProductList[i]);
+  }
```

Comment:
Alternative way would be to use `forEach()` method to iterate through the array of products

```
allProductList.forEach(singleProduct => {
     total += updateSubtotal(singleProduct)
});
```